common/types/nodeid.go

```diff
@@ -36,7 +37,7 @@ func (id NodeID) Bytes() []byte {
 
 // ShortString returns a the first 5 characters of the ID, for logging purposes.
 func (id NodeID) ShortString() string {
-	return Shorten(id.String(), 5)
+	return fmt.Sprintf("%.5s", id.String())
```

Comment:
Perhaps, as `ShortString()` is called quite often and usually eagerly (for logging), it could avoid encoding all 32B and then cut it to 5 characters. It could encode only the first 3B and reslice the string instead of using `Sprintf`.

```suggestion
	return hex.EncodeToString(id[:3])[:5]
```